### DISCUSSION TITLE
Add TLS support for MQTT

### DISCUSSION THREAD

**Discussion by git-developer** [BlackZork/mqmgateway#51] [Ideas]
Currently, the MQTT connection to the broker is plain / unencrypted. I'd like to propose to add TLS support for secure connections.

Motivation for this feature is obvious: adding security to communication. It's currently possible to connect using credentials (username / password). Transport of these without encryption imposes a security risk.

The underlying MQTT library mosquitto already supports TLS, so it's probably a matter of extending the configuration and delegating the TLS settings to mosquitto.

I'm aware that for fully-fledged TLS support, a bunch of settings might need to be adopted. My use case is simple and could be a point to start with. I'd like to give a CA certificate file to mosquitto so that the communication is encrypted. No need for a client certificate, a custom TLS version or the like.

**Comment by BlackZork** [comment]
> The underlying MQTT library mosquitto already supports TLS, so it's probably a matter of extending the configuration and delegating the TLS settings to mosquitto.

Yes, I would use the same configuration key names as int `mosquitto_sub --help` output. You may just add what you need.

**Comment by git-developer** [comment]
This feature was added in v2.8.0
